src/Place/Models/PDB01MPlace/Contexts/PCT011MPlacesList/MPlacesList.tsx

```diff
@@ -0,0 +1,119 @@
+import {
+	collection,
+	query,
+	where,
+	getDocs,
+	orderBy,
+	Query,
+	QueryConstraint,
+	startAfter,
+	limit,
+	DocumentSnapshot,
+} from 'firebase/firestore';
+import { useState, createContext, useEffect, ReactNode } from 'react';
+
+import {
+	MPlacesListInterface,
+	MPlacesListValInterface,
+	MPlacesListAddressInterface,
+} from 'spelieve-common/lib/Interfaces/Place';
+import { MPlace } from 'spelieve-common/lib/Models/Place/PDB01/MPlace';
+import { FirestoreConverter } from 'spelieve-common/lib/Utils/FirestoreConverter';
+
+import db from '@/Place/Endpoint/firestore';
+
+export const PCT011MPlacesList = createContext({} as MPlacesListValInterface);
+
+export const PCT011MPlacesListProvider = ({ children }: { children: ReactNode }) => {
+	const placeCollectionRef = collection(db, MPlace.modelName);
+	const [placesList, setPlacesList] = useState<MPlacesListInterface[]>([]);
+	const [isFirstLoading, setIsFirstLoading] = useState<boolean>(true);
+	const [address, setAddress] = useState<MPlacesListAddressInterface>({});
+	const [lastVisible, setLastVisible] = useState<DocumentSnapshot | null>(null);
+	const [isRefreshing, setIsRefreshing] = useState<boolean>(false);
+
+	const createBasicQueryConstraint = (): QueryConstraint[] => {
+		const qc: QueryConstraint[] = [];
+		qc.push(where(MPlace.Cols.country, '==', address.country));
+
+		if (address.administrativeAreaLevel1 !== '') {
+			// index済
+			qc.push(where(MPlace.Cols.administrativeAreaLevel1, '==', address.administrativeAreaLevel1));
+		}
+
+		if (address.administrativeAreaLevel2 !== '') {
+			qc.push(where(MPlace.Cols.administrativeAreaLevel2, '==', address.administrativeAreaLevel2));
+		}
+
+		if (address.locality !== '') {
+			// index済
+			qc.push(where(MPlace.Cols.locality, '==', address.locality));
+		}
+
+		qc.push(orderBy(MPlace.Cols.rating, 'desc'));
+		return qc;
+	};
+
+	const fetchSetPlaces = async (q: Query<MPlacesListInterface>, currentPlacesList: MPlacesListInterface[]) => {
+		await getDocs(q)
+			.then((querySnapshot) => {
+				setPlacesList([...currentPlacesList, ...querySnapshot.docs.map((doc) => doc.data())]);
+				setLastVisible(querySnapshot.docs[querySnapshot.docs.length - 1]);
+			})
```

Comment:
isFirstLoading と、isRefreshingがあってややこしいから
isLoadingだけにして、
```suggestion
	const fetchSetPlaces = async (q: Query<MPlacesListInterface>, currentPlacesList: MPlacesListInterface[]) => {
		await getDocs(q)
			.then((querySnapshot) => {
				setPlacesList([...currentPlacesList, ...querySnapshot.docs.map((doc) => doc.data())]);
				setLastVisible(querySnapshot.docs[querySnapshot.docs.length - 1]);
				setIsLoading(false);
			})
```
にすれば、fetchAdditionalPlacesもfetchFirstPlacesも消せてすごくシンプルになると思う！

Reply:
@Ayato-kosaka 
Places.tsxでisFirstLoadingを使用していて、PlacesList.tsxでisRefreshingを制御していたから
isLoadingにまとめると, 無限レンダリングの時（PlacesListを下スクロールした時）にもPlaces.tsx全体の再レンダリングがされちゃうって困っているんだけど、なんかいい方法あるかな

Reply:
@Takapy1 

PlacesList.tsxだけに isLoading持つか、
Places.tsx にローカルステートでisLoading作って、PlacesをuseEffectで監視して、[]だったらloading=trueとかでもいいかと、


Contextの責務はfetch する事で、isLoading は、fetch中を表す。
Page は描画することが責務で、isLoading は、Placesの値が入っていないことを表す。

みたいな整理かな、

Reply:
[d6bc736](https://github.com/Ayato-kosaka/spelieve/pull/173/commits/d6bc7361f08dfc6989807b0280aa32fbc20e9824)で対応’

Reply:
[5a84647](https://github.com/Ayato-kosaka/spelieve/pull/173/commits/5a84647d37c8cf5ba16f2b54f86452b936d6d9f8)で対応